dask/graph_manipulation.py

```diff
@@ -0,0 +1,426 @@
+        } | layer_deps_to_omit
+        if is_bound:
+            new_deps[new_layer_name].add(blocker_key)
+
+    # Add the layers of the collections from omit from child.dsk. Note that, when
+    # assume_layers=False, it would be unsafe to simply do HighLevelGraph.merge(dsk,
+    # omit[i].dsk). Also, collections in omit may or may not be parents of this specific
+    # child, or of any children at all.
+    while layers_to_copy_verbatim:
+        layer_name = layers_to_copy_verbatim.pop()
+        if layer_name in new_layers:
+            continue
+        layer_deps = dsk.dependencies[layer_name]
+        layers_to_copy_verbatim |= layer_deps
+        new_deps[layer_name] = layer_deps
+        new_layers[layer_name] = dsk.layers[layer_name]
+
+    rebuild, args = child.__dask_postpersist__()
+    return rebuild(
+        HighLevelGraph(new_layers, new_deps),
+        *args,
+        name=clone_key(prev_coll_name, seed),
+    )
+
+
+def clone(*collections, omit=None, seed: Hashable = None, assume_layers: bool = True):
+    """Clone dask collections, returning equivalent collections that are generated from
+    independent calculations.
+
+    Example
+    -------
+    (tokens have been simplified for the sake of brevity)
+
+    >>> from dask import array as da
+    >>> x_i = da.asarray([1, 1, 1, 1], chunks=2)
+    >>> y_i = x_i + 1
+    >>> z_i = y_i + 2
+    >>> dict(z_i.dask)
+    {('array-1', 0): array([1, 1]),
+     ('array-1', 1): array([1, 1]),
+     ('add-2', 0): (<function operator.add>, ('array-1', 0), 1),
+     ('add-2', 1): (<function operator.add>, ('array-1', 1), 1),
+     ('add-3', 0): (<function operator.add>, ('add-2', 0), 1),
+     ('add-3', 1): (<function operator.add>, ('add-2', 1), 1)}
+    >>> w_i = clone(z_i, omit=x_i)
+    >>> w_i.compute()
+    array([4., 4., 4., 4.])
+    >>> dict(w_i.dask)
+    {('array-1', 0): array([1, 1]),
+     ('array-1', 1): array([1, 1]),
+     ('add-4', 0): (<function operator.add>, ('array-1', 0), 1),
+     ('add-4', 1): (<function operator.add>, ('array-1', 1), 1),
+     ('add-5', 0): (<function operator.add>, ('add-4', 0), 1),
+     ('add-5', 1): (<function operator.add>, ('add-4', 1), 1)}
+
+    Parameters
+    ----------
+    collections
+        Zero or more Dask collections or nested structures of Dask collections
+    omit
+        Dask collection or nested structure of Dask collections which will not be cloned
+    seed
+        See :func:`bind`
+    assume_layers
+        See :func:`bind`
+
+    Returns
+    -------
+    Same as ``collections``
+        Dask collections of the same type as the inputs, which compute to the same
+        value, or nested structures equivalent to the inputs, where the original
+        collections have been replaced.
+    """
+    out = bind(
+        collections, parents=None, omit=omit, seed=seed, assume_layers=assume_layers
+    )
+    return out[0] if len(collections) == 1 else out
+
+
+def block_until_done(*collections):
```

Comment:
Thinking about this again, what do you about `wait_on` for this method instead? It's shorter and doesn't use the word "block" which might lead to confusion about _when_ the blocking occurs (Apologies for suggesting a name change so late in the process)

Reply:
fine for me; I thought you wanted to avoid the word "wait" as it may lead to confusion with asyncio and blocking calls